pylhc/autosix.py

```diff
@@ -0,0 +1,525 @@
+        messages are logged to debug).
+        > cd $basedir/workspace-$jobname/sixjobs
+        > /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/bash/run_status
+
+        If not, and resubmit is active
+        > cd $basedir/workspace-$jobname/sixjobs
+        > /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/bash/run_six.sh -i
+        """
+        if check_ok:
+            check_sixtrack_output(jobname, basedir, ssh=ssh, resubmit=resubmit)
+
+    with check_stage(STAGES.sixdb_load, jobname, basedir) as check_ok:
+        """
+        Gather results into database via sixdb.
+        > cd $basedir/workspace-$jobname/sixjobs
+        > python3 /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/externals/SixDeskDB/sixdb . load_dir
+        """
+        if check_ok:
+            sixdb_load(jobname, basedir, python=python, ssh=ssh)
+
+    with check_stage(STAGES.sixdb_cmd, jobname, basedir) as check_ok:
+        """
+        Analysise results in database via sixdb.
+        > cd $basedir/workspace-$jobname/sixjobs
+        > python3 /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/externals/SixDeskDB/sixdb $jobname da
+
+        when fixed:
+        > python3 /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/externals/SixDeskDB/sixdb $jobname da_vs_turns -turnstep 100 -outfile
+        > python3 /afs/cern.ch/project/sixtrack/SixDesk_utilities/pro/utilities/externals/SixDeskDB/sixdb $jobname plot_da_vs_turns
+        """
+        if check_ok:
+            sixdb_cmd(jobname, basedir, cmd=["da"], python=python, ssh=ssh)
+
+            # da_vs_turns is broken at the moment (jdilly, 19.10.2020)
+            # sixdb_cmd(jobname, basedir, cmd=['da_vs_turns', '-turnstep', str(da_turnstep), '-outfile'],
+            #           python=python, ssh=ssh)
+            # sixdb_cmd(jobname, basedir, cmd=['plot_da_vs_turns'], python=python, ssh=ssh)
+
+    with check_stage(STAGES.post_process, jobname, basedir) as check_ok:
+        """
+        Extracts the analysed data in the database and writes them to three tfs files:
+
+        - All DA values
+        - Statistics over angles, listed per seed (+ Seed 0 as over seeds and angles)
+        - Statistics over seeds, listed per angle
+
+        The statistics over the seeds are then plotted in a polar plot.
+        All files are outputted to the ``sixjobs/autosix_output`` folder in the job directory.
+        """
+        if check_ok:
+            post_process_da(jobname, basedir)
+
+    with check_stage(STAGES.final, jobname, basedir) as check_ok:
+        """ Just info about finishing this script and where to check the stagefile. """
+        if check_ok:
+            stage_file = get_stagefile_path(jobname, basedir)
+            LOG.info(
+                f"All stages run. Check stagefile {str(stage_file)} "
+                "in case you want to rerun some stages."
+            )
+            raise StageSkip()
```

Comment:
I don't get raising an error here?

Reply:
that's not really an error, that is the way to skip the stage (not declare it finshed). And I want this stage to always be re-run as it checks if all other stages have run and prints it out for the user to read